### PR TITLE
feat: trpc v10 migration

### PR DESCRIPTION
**Pull request by juliusmarminge**
## What does this PR do?

Updates tRPC to v10

Fixes # (issue)

**Environment**: Staging(main branch) / Production

## Type of change

- [x] Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

- [ ] Test A
- [ ] Test B

## Checklist

- I haven't commented my code, particularly in hard-to-understand areas
- I haven't checked if new and existing unit tests pass locally with my changes
  - ~~some jest mocking still to resolve~~ we just put a skip on here for now

### CODE REVIEW

**Comment by juliusmarminge** [comment]
Blocked by https://github.com/trpc/trpc/pull/3108

File: packages/trpc/server/routers/public.ts
Comment (juliusmarminge) [inline packages/trpc/server/routers/public.ts:3]: moved into the "real" publicRouter

File: packages/trpc/server/routers/viewer/availability.tsx
Comment (juliusmarminge) [inline packages/trpc/server/routers/viewer/availability.tsx:44]: since we have a router called schedule, we cannot have a procedure named the same, so this procedure is now schedule.get

File: packages/trpc/server/routers/viewer/bookings.tsx
Comment (juliusmarminge) [inline packages/trpc/server/routers/viewer/bookings.tsx:312]: instead of nested middlewares, we're using multiple input parsers: https://trpc.io/docs/v10/procedures#multiple-input-parsers

File: apps/web/components/ui/NoCalendarConnectedAlert/__tests__/NoCalendarConnectedAlert.test.tsx
Comment (KATT) [inline apps/web/components/ui/NoCalendarConnectedAlert/__tests__/NoCalendarConnectedAlert.test.tsx:7]: `skip` here maybe

**Comment by PeerRich** [comment]
CleanShot 2022-11-04 at 20 20 02@2x


idk what happened but a ton of merge conflicts @KATT @juliusmarminge @zomars

**Comment by juliusmarminge** [comment]
I can resolve them now if you want, but they will keep coming unless it's merged soon after ;)

**Comment by juliusmarminge** [comment]
serum why is rollup complaining 🤔

**Comment by juliusmarminge** [comment]
doesn't seem to be just this PR though

https://github.com/calcom/cal.com/actions/runs/3396873456/jobs/5648421812

**Comment by KATT** [comment]
Merge conflicts are expected to be piling up quickly as things are merged as this touches so much of the application, so it's best if we do a swift cycle of Review -> Fix conflicts -> Re-review -> Merge

File: apps/web/components/ui/NoCalendarConnectedAlert/__tests__/NoCalendarConnectedAlert.test.tsx
Comment (KATT) [inline apps/web/components/ui/NoCalendarConnectedAlert/__tests__/NoCalendarConnectedAlert.test.tsx:8]: This might help https://github.com/trpc/trpc/discussions/1879#discussioncomment-3846341

**Comment by PeerRich** [comment]
> Merge conflicts are expected to be piling up quickly as things are merged as this touches so much of the application, so it's best if we do a swift cycle of Review -> Fix conflicts -> Re-review -> Merge

agree, sorry about this. lets get this merged tomorrow.🙏 @zomars can help

**Comment by KATT** [comment]
> agree, sorry about this. lets get this merged tomorrow.🙏 @zomars can help

All good! :) 

Ideally, we only deal with merge conflicts _once_ just before merging (if more stuff gets merged ahead of this one, that is). 

I mainly want to minimize the number of times @juliusmarminge or I will get it up-to-date with `main`, but please take your time and be diligent with the review of this PR since it's a big change & we don't know what level of confidence you have in the automatic test suite 😊

File: packages/trpc/server/routers/viewer/webhook.tsx
Comment (KATT) [inline packages/trpc/server/routers/viewer/webhook.tsx:30]: I think this could be simplified a bit by splitting it up into 2 base procedures.

- One can use [multiple input parsers](https://trpc.io/docs/v10/procedures#multiple-input-parsers) with the `webhookIdAndEventTypeIdSchema`
- One can be without

**Comment by juliusmarminge** [comment]
> lets get this merged tomorrow.

I'm available to resolve conflicts all day if you get this reviewed 👍🏼

File: packages/trpc/server/routers/viewer/webhook.tsx
Comment (juliusmarminge) [reply packages/trpc/server/routers/viewer/webhook.tsx:30]: Would you mind looking at that?

I don't really understand how I'd split this up since their endpoints mostly use optional input and putting the multiple input parsers would make all of these "required", i.e. not using the same middleware but early returning if the input isn't defined?

On the one hand, if they don't pass any input it's just an authedProcedure, but looks like even the `webhook.list` optionally use the input (which should in that case go through the middleware)

File: apps/web/pages/availability/index.tsx
Comment (juliusmarminge) [inline apps/web/pages/availability/index.tsx:27]: First arg is now the `input` which can be escaped by passing undefined if there isn't one

File: packages/features/ee/workflows/components/WorkflowDetailsPage.tsx
Comment (juliusmarminge) [inline packages/features/ee/workflows/components/WorkflowDetailsPage.tsx:98]: Got type-error here saying this was required so threw in a null there

File: packages/trpc/server/routers/viewer/webhook.tsx
Comment (KATT) [reply packages/trpc/server/routers/viewer/webhook.tsx:30]: It's an non-need optimization, let's get this merged first and then I can bring some ideas to the team on how to improve stuff! :)

**Comment by zomars** [comment]
Conflicts resolved. Reviewing...

File: apps/web/components/integrations/CalendarListContainer.tsx
Comment (zomars) [inline apps/web/components/integrations/CalendarListContainer.tsx:138]: What does this `trpc` prop doing in this context?

Comment (zomars) [reply apps/web/components/integrations/CalendarListContainer.tsx:138]: cc @KATT

File: apps/web/components/ui/NoCalendarConnectedAlert/__tests__/NoCalendarConnectedAlert.test.tsx
Comment (zomars) [reply apps/web/components/ui/NoCalendarConnectedAlert/__tests__/NoCalendarConnectedAlert.test.tsx:8]: @alannnc can you take a look?

File: apps/web/pages/event-types/index.tsx
Comment (zomars) [inline apps/web/pages/event-types/index.tsx:99]: Good enough for now

File: packages/app-store/BookingPageTagManager.tsx
Comment (zomars) [inline packages/app-store/BookingPageTagManager.tsx:40]: Unrelated change to migration, will revert.

File: packages/features/ee/workflows/pages/v2/workflow.tsx
Comment (zomars) [inline packages/features/ee/workflows/pages/v2/workflow.tsx:14]:
```suggestion
```

**Review by zomars — APPROVED** [review]
Impressive work @juliusmarminge tested locally and all seems to work. Will wait for checks to pass

LGTM

**Comment by KATT** [comment]
giphy

File: packages/app-store/BookingPageTagManager.tsx
Comment (juliusmarminge) [reply packages/app-store/BookingPageTagManager.tsx]: Sorry bout that one. Not sure why that got removed 🤔

File: apps/web/components/integrations/CalendarListContainer.tsx
Comment (juliusmarminge) [reply apps/web/components/integrations/CalendarListContainer.tsx:138]: It is for tRPC related options (skipBatch, abortOnUnmount, ssr etc). 

The codemod "accidentally" added this prop even if it's empty but i got rid of most of them (this one must have slipped through).

**Pull request merged**
